### ISSUE TITLE
RSA normal file write operation output to stderr instead of stdout

### ISSUE THREAD

**Issue by foxylad** [openssl/openssl#22085]
First up, thanks for your awesome work.

I've been automating submitting updated certificates to Google Appengine, which expects the private key in RSA format. I use `openssl rsa -in _filename_ -out _filename.rsa_` to achieve this. Unfortunately openssl writes "writing RSA key" to stderr which makes my script think something is wrong when it isn't.

I think this is a 1 minute fix. Line 316 of `/apps/rsa.c` currently reads:
```
    BIO_printf(bio_err, "writing RSA key\n");
```
It should read:
```
    BIO_printf(out, "writing RSA key\n");
```
Or possibly remove the line altogether - it doesn't add much information.

**Comment by slontis** [comment]
I think before fixing this we need to figure out what the intention was for all OpenSSL applications in relation to output...

bio_in is for stdin
bio_err is stderr.
bio_out is stdout.

Applications use params such as -out as a target file to write data to (in this case the rsa key).
If you dont specify -out is uses stdout
So I assume this app chose stderr so it didnt get written to the target file.

I would assume that in most cases we would want to use bio_out for this case..
Although I can see that maybe tacking additional data onto the end of a PEM file might be useful? (DSA key gen parameters comes to mind).
@levitte any thoughts on this?

**Comment by nhorman** [comment]
It seems to me that a calling script shouldn't use non-empty stderr output to imply an actual error at all, but rather the exit code of the application.

Regardless, since this has gone un-commented for a year, I'm marking it as inactive, to be closed at the end of the 3.4 dev cycle without further input/updates
